### PR TITLE
Added Shortest Job First Algorithm

### PR DESCRIPTION
**Pull request by pharshil1902**

### CODE REVIEW

**Comment by mrmaxguns** [comment]
You got an error with the doctest:
This function calculates the average of the waiting & turnaround times
```
092     >>> calculate_average_times([0,3,5,0],[3,6,10,1],4)
Expected:
    Average waiting time = 2.00000 
    Average turn around time =  5.0

Got:
    <BLANKLINE>                <------------------- The culprit
    Average waiting time = 2.00000 
    Average turn around time =  5.0
```
It looks like your code is returning a blank line. This should be an easy fix.

File: scheduling/shortest_job_first_algorithm.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+# Shortest job remainig first
+# PLease note that arriaval time and burst 
+#time are inputted as space separated
+
+import pandas as pd
+
+       
+
+def calculate_waitingtime(arrival_time,burst_time,no_of_processes):
```
Comment (cclauss) [inline scheduling/shortest_job_first_algorithm.py:11]: https://docs.python.org/3/library/typing.html please.

File: scheduling/shortest_job_first_algorithm.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+   
+    complete = 0
+    increment_time = 0
+    minm = 999999999
+    short = 0
+    check = False
+
+    # Process until all processes gets 
+    # completed 
+    while (complete != no_of_processes):
```
Comment (cclauss) [inline scheduling/shortest_job_first_algorithm.py:37]:
```suggestion
    while complete != no_of_processes:
```

File: scheduling/shortest_job_first_algorithm.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        for j in range(no_of_processes):
+            if(arrival_time[j]<=increment_time):
+                if(remaining_time[j]>0):
+                    if(remaining_time[j]<minm):
```
Comment (cclauss) [inline scheduling/shortest_job_first_algorithm.py:41]:
```suggestion
            if arrival_time[j]<=increment_time:
                if remaining_time[j]>0:
                    if remaining_time[j]<minm:
```

File: scheduling/shortest_job_first_algorithm.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                        minm=remaining_time[j]
+                        short=j
+                        check=True
+            
+        if (check == False):
```
Comment (cclauss) [inline scheduling/shortest_job_first_algorithm.py:46]:
```suggestion
        if not check: 
```

File: scheduling/shortest_job_first_algorithm.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+            finar=finish_time - arrival_time[short]
+            waiting_time[short] = (finar - burst_time[short]) 
+
+            if (waiting_time[short] < 0): 
+                waiting_time[short] = 0
+        
+        # Increment time 
+        increment_time += 1
+    return waiting_time
+def calculate_turnaroundtime(burst_time, no_of_processes, waiting_time):
```
Comment (cclauss) [inline scheduling/shortest_job_first_algorithm.py:73]: https://docs.python.org/3/library/typing.html please.

File: scheduling/shortest_job_first_algorithm.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    >>> calculate_turnaroundtime([3,3], 2, [0,3])
+    [3, 6]
+    >>> calculate_turnaroundtime([8,10,1], 3, [1,0,3])
+    [9, 10, 4]
+    """
+    turn_around_time=[0]*no_of_processes
+    for i in range(no_of_processes): 
+        turn_around_time[i] = burst_time[i] + waiting_time[i] 
+    return turn_around_time
+def calculate_average_times(waiting_time,turn_around_time, no_of_processes):
```
Comment (cclauss) [inline scheduling/shortest_job_first_algorithm.py:88]: https://docs.python.org/3/library/typing.html please.

**Comment by cclauss** [comment]
Please stop closing and opening new PRs.  All the comments are lost.  Just keep the PR open and edit the files.  We will get there and when we do, we will "squash and merge" which will squeeze all the commits into a single commit.

**Pull request closed without merging**